src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php

```diff
@@ -20,6 +20,7 @@
 /**
  * @template TRelatedModel of \Illuminate\Database\Eloquent\Model
  * @template TDeclaringModel of \Illuminate\Database\Eloquent\Model
+ * @template TPivotModel of \Illuminate\Database\Eloquent\Relations\Concerns\AsPivot
```

Comment:
Traits can't be used as types, so this should probably be:
```suggestion
 * @template TPivotModel of \Illuminate\Database\Eloquent\Relations\Pivot
```

Reply:
hmm... does this makes sense then even when the called static method is specifically on the Trait?

Reply:
I guess it does, because you're supposed to extend `Pivot` for your custom pivot models.

Thanks for the help!